Source/Core/Core/PowerPC/GDBThread.cpp

```diff
@@ -0,0 +1,1274 @@
+  }
+
+  NOTICE_LOG(GDB_THREAD, "Terminating GDB stub thread.");
+}
+
+// private helpers
+static u8 HexToChar(u8 hex)
+{
+  if (hex >= '0' && hex <= '9')
+    return hex - '0';
+  else if (hex >= 'a' && hex <= 'f')
+    return hex - 'a' + 0xa;
+  else if (hex >= 'A' && hex <= 'F')
+    return hex - 'A' + 0xa;
+
+  DEBUG_LOG(GDB_THREAD, "Invalid nibble: %c (%02x)\n", hex, hex);
+  return 0;
+}
+
+static u8 NibbleToHex(u8 n)
+{
+  n &= 0xf;
+  if (n < 0xa)
+    return '0' + n;
+  else
+    return 'A' + n - 0xa;
+}
+
+static void MemToHex(u8* dst, u32 src, u32 len)
+{
+  while (len-- > 0)
+  {
+    u8 tmp = PowerPC::HostRead_U8(src++);
+
+    *dst++ = NibbleToHex(tmp >> 4);
+    *dst++ = NibbleToHex(tmp);
+  }
+}
+
+static void HexToMem(u32 dst, u8* src, u32 len)
+{
+  while (len-- > 0)
+  {
+    u8 tmp = HexToChar(*src++) << 4;
+    tmp |= HexToChar(*src++);
+
+    PowerPC::HostWrite_U8(tmp, dst++);
+  }
+}
+
+static void WriteHexBE32(u8* p, u32 v)
+{
+  for (u32 i = 0; i < 8; i++)
+    p[i] = NibbleToHex(u8(v >> (28 - 4 * i)));
+}
+
+static void WriteHexBE64(u8* p, u64 v)
+{
+  for (u32 i = 0; i < 16; i++)
+    p[i] = NibbleToHex(u8(v >> (60 - 4 * i)));
+}
+
+static u32 ReadHexBE32(const u8* p)
+{
+  u32 res = 0;
+
+  for (u32 i = 0; i < 8; i++)
+    res = (res << 4) | HexToChar(p[i]);
+
+  return res;
+}
+
+static u64 ReadHexBE64(const u8* p)
+{
+  u64 res = 0;
+
+  for (u32 i = 0; i < 16; i++)
+    res = (res << 4) | HexToChar(p[i]);
+
+  return res;
+}
+
+// GDB stub interface
+#ifndef _WIN32
+void GDBThread::GDBInitLocal(const char* socket)
+{
+  unlink(socket);
+
+  sockaddr_un addr = {};
+  addr.sun_family = AF_UNIX;
+  strcpy(addr.sun_path, socket);
+
+  GDBInitGeneric(PF_LOCAL, reinterpret_cast<const sockaddr*>(&addr), sizeof(addr), nullptr,
+                 nullptr);
+}
+#endif
+
+void GDBThread::GDBInit(u32 port)
+{
+  GDBDeinit();
+
+  sockaddr_in saddr_server = {};
+  sockaddr_in saddr_client;
+
+  saddr_server.sin_family = AF_INET;
+  saddr_server.sin_port = htons(port);
+  saddr_server.sin_addr.s_addr = INADDR_ANY;
+
+  socklen_t client_addrlen = sizeof(saddr_client);
+
+  GDBInitGeneric(PF_INET, reinterpret_cast<const sockaddr*>(&saddr_server), sizeof(saddr_server),
+                 reinterpret_cast<sockaddr*>(&saddr_client), &client_addrlen);
+
+  saddr_client.sin_addr.s_addr = ntohl(saddr_client.sin_addr.s_addr);
+}
+
+void GDBThread::GDBInitGeneric(int domain, const sockaddr* server_addr, socklen_t server_addrlen,
+                               sockaddr* client_addr, socklen_t* client_addrlen)
+{
+#ifdef _WIN32
+  WSADATA init_data;
+  WSAStartup(MAKEWORD(2, 2), &init_data);
+#endif
+
+  m_tmpsock = socket(domain, SOCK_STREAM, 0);
+  if (m_tmpsock == -1)
+    ERROR_LOG(GDB_THREAD, "Failed to create gdb socket");
+
+  int on = 1;
+  if (setsockopt(m_tmpsock, SOL_SOCKET, SO_REUSEADDR, reinterpret_cast<const char*>(&on),
+                 sizeof on) < 0)
+    ERROR_LOG(GDB_THREAD, "Failed to setsockopt");
```

Comment:
In case this one sticks around as a one-liner, it should still be braced (since the condition spans over two lines)

Reply:
Inherited from Dolphin's GDBStub implementation. Was originally one line until previous comment suggested updating to reinterpret_cast, which caused auto formatting to push it to second line. Will take a look.